WordPress/openverse-frontend
# Search and media store refactoring

## Fixes


Fixes #383 by @zackkrida

## Description


This is the first part of another big store refactoring. Instead of a `search` and `filter` store, there are a `media` and a `search` stores.
### Media store
This store is responsible for keeping information about the search results: the AudioResult or ImageResult arrays, the current Audio or Image, as well as the data about the fetching state (whether it is fetching or not, whether there is a fetching error).
### Search store
This is where all the information about applied filters and search query is kept.

I've also added docstrings to actions, and found that they didn't have the necessary `await`s in them, which caused problems with loading (I still need to add several docstring descriptions).

### Next steps
The next step should be simplifying the query and filter handling:
The filters should be the source of truth for all query parameters except for `q` and `mediaType`. This would also simplify media fetching for different media types.

## Testing Instructions

You can run the unit tests. Another thing I did was to create a new branch, merge the e2e tests from #394 into it, and test with `npm run test:e2e`. This has helped me see bugs that I didn't realize exist when testing manually. They all currently pass.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
